app/Models/Posts/Authors.php

```diff
@@ -15,80 +16,31 @@ class Authors extends Model
 
     public function posts()
     {
-        return $this->hasMany('App\Models\Posts');
+        return $this->hasMany('App\Models\Posts', 'id', 'author_id');
     }
     
     
-    // @codeCoverageIgnoreStart
-    
-    // Accessor to get full URL of image
-    public function getImageAttribute()
-    {
-        $disk = 'images';
-        $filename = $this->attributes['image'];
-        
-        return $filename ? self::public_url($disk, $filename) : null;
-    }
-    
-    
-    // Mutator to upload an image
     public function setImageAttribute($value)
     {
-        $attribute_name = "image";
-        $disk = "images";
-        $destination_path = "images";
+        $attribute_name = 'image';
+        $disk = 'authors_images';
+        $destination_path = '';
 
-        // if the image was erased
-        if ($value==null) {
-            // delete the image from disk
-            \Storage::disk($disk)->delete($this->{$attribute_name});
-
-            // set null in the database column
-            $this->attributes[$attribute_name] = null;
-        }
-
-        // if a base64 was sent, store it in the db
-        if (starts_with($value, 'data:image')) {
-            // 0. Make the image
-            $image = \Image::make($value);
-            // 1. Generate a filename.
-            $filename = md5($value.time()).'.jpg';
-            // 2. Store the image on disk.
-            \Storage::disk($disk)->put($destination_path.'/'.$filename, $image->stream());
-            // 3. Save the path to the database
-            $this->attributes[$attribute_name] = /*$destination_path.'/'.*/$filename;
-        }
+        $this->uploadFileToDisk($value, $attribute_name, $disk, $destination_path);
     }
     
-    
-    
-    // Get a public URL from a path
-    // Stolen from https://stackoverflow.com/questions/37017880/laravel-filesystem-storage-local-public-file-urls-not-working#comment61589587_37018544
-    protected function public_url($disk, $path = '')
+    public static function boot()
     {
-        $fs = \Storage::disk($disk);
-
-        if ($fs->getAdapter() instanceof \League\Flysystem\Adapter\Local) {
-            return asset($fs->url($path));
-        }
-
-        return $fs->url($path);
-    }
-    
-    
-    // 'Custom' serializer to use accessor for image attribute
-    public function toJson($options=0)
-    {
-        $this->append('image');
-        return parent::toJson($options);
+        parent::boot();
+        static::deleting(function (Authors $author) {
+            if (isset($author->image)) {
+                \Storage::disk('authors_images')->detach($author->image);
+            }
+
+            Posts::where('author_id', '=', $author->id)->get()->each(function (Posts $p) {
+                $p->author()->associate(null);
```

Comment:
Isn't `dissociate()` a better solution for this?

Reply:
Fixed